Objective-C/Tests/URLEndpointListenerTest.m

```diff
@@ -245,4 +251,51 @@ - (void) testPaswordAuthenticator {
     [listener stop];
 }
 
+// Disable until completing change in CBLWebSocket:
+- (void) _testClientCertAuthenticatorWithBlock API_AVAILABLE(ios(10.3), macosx(10.5)) {
+    if (!self.keyChainAccessAllowed) return;
+    
+    // Listener:
+    CBLListenerCertificateAuthenticator* listenerAuth =
+    [[CBLListenerCertificateAuthenticator alloc] initWithBlock:
+    ^BOOL(NSArray *certs) {
+        SecCertificateRef cert = (__bridge SecCertificateRef)(certs[0]);
+        CFStringRef cnRef;
+        OSStatus status = SecCertificateCopyCommonName(cert, &cnRef);
+        AssertEqual(status, errSecSuccess);
+        NSString* cn = (NSString*)CFBridgingRelease(cnRef);
+        return [cn isEqualToString: @"daniel"];
+    }];
+    
+    Listener* listener = [self listenWithTLS: YES auth: listenerAuth];
+    AssertNotNil(listener);
+    
+    // Cleanup:
+    NSError* error;
+    Assert([CBLTLSIdentity deleteIdentityWithLabel: @"client1" error: &error]);
+    
+    // Create client identity:
+    NSDictionary* attrs = @{ kCBLCertAttrCommonName: @"daniel" };
+    CBLTLSIdentity* identity1 = [CBLTLSIdentity createIdentityForServer: NO
+                                                             attributes: attrs
+                                                             expiration: nil
+                                                                  label: @"client1"
+                                                                  error: &error];
+    AssertNotNil(identity1);
+    AssertNil(error);
+    
+    CBLReplicatorConfiguration* config = nil;
+    CBLClientCertificateAuthenticator* auth = nil;
+    auth = [[CBLClientCertificateAuthenticator alloc] initWithIdentity: identity1];
+    pinServerCert = NO;
+    config = [self configWithTarget: listener.localEndpoint
+                               type: kCBLReplicatorTypePushAndPull
+                         continuous: NO
+                      authenticator: auth];
+    [self run: config errorCode: 0 errorDomain: nil];
```

Comment:
validate replication, at least a document between listener and replicator?

Reply:
I think it’s optional as the replicator is connected and caught up.

Reply:
Agreed, as long as we capture all errors thrown.